### DISCUSSION TITLE
Connecting to HiveMQ

### DISCUSSION THREAD

**Discussion by leisurehound** [adam-fowler/mqtt-nio#142] [General]
Am attempting to connect to my HiveMQ broker but admit I'm not well versed on all all the TLS configurations I need.

```
    mqttConfig = MQTTClient.Configuration(
      pingInterval: .seconds(30),
      userName: username,
      password: password,
      useSSL: true,
      sniServerName: sniServerName)


  client = MQTTClient(host: url, port: port, identifier: deviceID, eventLoopGroupProvider: .createNew, configuration: mqttConfig)

  Task {
      do {
        _ = try await client.connect()
        client.addCloseListener(named: "rpitelem-config") {
          result in
          print("reached close listener")
        }
      } catch (let error) {
        print("Error on connect \(error.localizedDescription)")
        print("\(error)")
      }
    }
```

For sniServerName I'm using the TLS URI that is shown for my cluster.  I have confirmed the username/password are correct (they work from an old C implementation of what I'm building in Swift).

I get the following error being caught:
```
Error on connect The operation couldn’t be completed. (MQTTNIO.MQTTError error 0.)
connectionError(MQTTNIO.MQTTError.ConnectionReturnValue.notAuthorized)
```

I've tried various values for `tslConfiguration` but none with any success.

Years ago when I wrote  the C impl using the WolfMQTT library (and WolfSSL) and couldn't get it to work, WolfSSL mentioned that HiveMQ 'requires the client TLS handshake to supply the Server Name Indication extension' which I believe I am doing now in Swift with MQTTNIO, no?

Any advice to set the TLS correctly to connect to hiveMQ?  In the C impl, I don't believe I'm providing a certificate or a private key..

**Comment by adam-fowler** [comment]
You only need to provide the sniServerName if it is different from the server you are connecting to.

**Comment by leisurehound** [comment]
Thanks, the TLS URI is different than server name,  not setting that (/or setting it to the server name) allows me to connect.  Thanks for your reply.
